### PR TITLE
WIP Feature/2957 turn off language detection

### PR DESCRIPTION
**Pull request by amoedoamorim**
## Description

This PR adds a Language Detection toggle in the Language settings page along with a mutation to persist this setting.
Also the styles were updated in that page to match current designs and a few string descriptions were sneaked in too.

References: CV2-2957

## Type of change

- [x] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?

Tested manually that the mutation is sent and the value is persisted and updated on the page.

## Things to pay attention to during code review

Nothing special going on here. Just let's make sure we're following all current coding guidelines.

## Checklist

- [x] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [x] I have removed the /* eslint-disable @calm/react-intl/missing-attribute */ from any files I've worked on and added a `description` prop to all `<FormattedMessage />` components that were missing it.
- [x] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)

### CODE REVIEW

File: src/app/components/team/Languages/LanguagesComponent.js [hunk 1]
```diff
@@ -1,66 +1,142 @@
-/* eslint-disable @calm/react-intl/missing-attribute, relay/unused-fields */
 import React from 'react';
 import { PropTypes } from 'prop-types';
-import { FormattedMessage } from 'react-intl';
-import { createFragmentContainer, graphql } from 'react-relay/compat';
-import Card from '@material-ui/core/Card';
-import CardContent from '@material-ui/core/CardContent';
+import { FormattedMessage, FormattedHTMLMessage } from 'react-intl';
+import { createFragmentContainer, graphql, commitMutation } from 'react-relay/compat'; // eslint-disable-line
```
Comment (caiosba) [inline src/app/components/team/Languages/LanguagesComponent.js:4]: Why disable this line for the linter?

File: src/app/components/team/Languages/LanguagesComponent.js [hunk 2]
```diff
@@ -1,66 +1,142 @@
...
 import { ContentColumn } from '../../../styles/js/shared';
+import styles from './LanguagesComponent.module.css';
+
+const submitToggleLanguageDetection = ({
+  team,
+  value,
+  onSuccess,
+  onFailure,
+}) => {
+  console.log('value', value); // eslint-disable-line
```
Comment (caiosba) [inline src/app/components/team/Languages/LanguagesComponent.js:22]: Just a reminder to remove this.

File: src/app/components/team/Languages/LanguagesComponent.js [hunk 3]
```diff
@@ -1,66 +1,142 @@
...
 
 const LanguagesComponent = ({ team }) => {
   const defaultCode = team.get_language || 'en';
 
   let languages = safelyParseJSON(team.get_languages) || [];
   languages = languages.sort((a, b) => compareLanguages(defaultCode, a, b));
 
+  const toggleLanguageDetection = (value) => {
+    const onSuccess = () => { console.log('success'); }; // eslint-disable-line
+    const onFailure = () => { console.log('failure'); }; // eslint-disable-line
```
Comment (caiosba) [inline src/app/components/team/Languages/LanguagesComponent.js:58]: Reminder to refactor those

**Review by caiosba — CHANGES_REQUESTED** [review]
I commented a few things which I think are expected for a "draft" PR... please re-request review when it's ready.

File: src/app/components/team/TeamComponent.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-/* eslint-disable @calm/react-intl/missing-attribute, relay/unused-fields */
+/* eslint-disable relay/unused-fields */
```
Comment (dariusk) [inline src/app/components/team/TeamComponent.js:1]: :tada:

File: src/app/components/team/Languages/LanguagesComponent.js [hunk 5]
```diff
@@ -1,66 +1,148 @@
...
+      setFlashMessage((
+        getErrorMessageForRelayModernProblem(errors)
+        || <GenericUnknownErrorMessage />
+      ), 'error');
+    };
+
+    submitToggleLanguageDetection({
+      team,
+      value,
+      onSuccess: () => {},
```
Comment (dariusk) [inline src/app/components/team/Languages/LanguagesComponent.js:69]: I see the design has nothing happening on success here, so I guess this is not a big problem but want to flag for @brianfleming that we probably want to figure out some kind of generic "success" indicator for all our invisible inputs where you flip a switch and we quickly talk to check-api to toggle something.

**Review by caiosba — APPROVED** [review]

**Pull request merged**